packages/frontend/src/components/staking/components/Validator.js

```diff
@@ -21,20 +27,41 @@ export default function Validator({
     currentValidators,
 }) {
     const [confirm, setConfirm] = useState(null);
+    const [farmList, setFarmList] = useState([]);
+    const nearAsFT = useNEARAsTokenWithMetadata();
+    const accountId = useSelector(selectAccountId);
+    const contractMetadata = useSelector(selectAllContractMetadata);
+
     const dispatch = useDispatch();
     const stakeNotAllowed = !!selectedValidator && selectedValidator !== match.params.validator && !!currentValidators.length;
     const showConfirmModal = confirm === 'withdraw';
 
+    useEffect(() => {
+        const getFarms = async () => {
+            const farms = await validator.contract.get_farms({from_index: 0, limit: 300});
+            setFarmList(await Promise.all(farms.map(({ token_id }, i) => {
+                dispatch(fetchToken({contractName: token_id}));
+                return validator.contract
+                    .get_unclaimed_reward({ account_id: accountId, farm_id: i })
+                    .catch(() => "0")
```

Comment:
`catch` is used because `get_unclaimed_reward` [will throw](https://github.com/referencedev/staking-farm/blob/master/staking-farm/src/views.rs#L148) when the account has not accrued any rewards with the contract yet

Reply:
Follow-up: error state; if we get a network request error (or really, anything other than `ERR_NO_ACCOUNT` or `ERR_NO_FARM` type errors), we shouldn't be showing the user a 0 value here.  This will need some thought from Corwin on design.

Reply:
Makes sense. For now I think this call silently failing unless we have data is fine (default balance is 0).